bn-apps/cordapp-updates-distribution/corda-updates-shell/src/main/kotlin/net/corda/businessnetworks/cordaupdates/shell/ConsoleTransferListener.kt

```diff
@@ -0,0 +1,123 @@
+package net.corda.businessnetworks.cordaupdates.shell
+
+import org.eclipse.aether.transfer.AbstractTransferListener
+import org.eclipse.aether.transfer.MetadataNotFoundException
+import org.eclipse.aether.transfer.TransferEvent
+import org.eclipse.aether.transfer.TransferResource
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import java.text.DecimalFormat
+import java.text.DecimalFormatSymbols
+import java.util.*
+import java.util.concurrent.ConcurrentHashMap
+
+/**
+ * Simple maven resolver transfer listener that logs everything into the console
+ * The implementation is kindly borrowed from https://github.com/eclipse/aether-demo/blob/master/aether-demo-snippets/src/main/java/org/eclipse/aether/examples/util/ConsoleTransferListener.java
+ */
+class ConsoleTransferListener : AbstractTransferListener() {
+    companion object {
+        val logger : Logger by lazy { LoggerFactory.getLogger("corda-updates") }
+    }
+    private val downloads = ConcurrentHashMap<TransferResource, Long>()
+    private var lastLength : Int = 0
+
+
+    override fun transferInitiated(event : TransferEvent) {
+        val message = when (event.requestType) {
+            TransferEvent.RequestType.PUT -> "Uploading"
+            else -> "Downloading"
+        }
+        logger.info(message + ": " + event.resource.repositoryUrl + event.resource.resourceName)
+    }
+
+    override fun transferProgressed(event : TransferEvent) {
+        val resource = event.resource
+        downloads[resource] = event.transferredBytes
+
+        val buffer = StringBuilder(64)
+
+        downloads.forEach {
+            val total = it.key.contentLength
+            buffer.append(getStatus(it.value, total)).append("  ")
+        }
+
+        val pad = lastLength - buffer.length
+        lastLength = buffer.length
+        pad(buffer, pad)
+        buffer.append('\r')
+
+        logger.info(buffer.toString())
```

Comment:
If you want a good progress bar library for console use, this one works well:

https://github.com/ctongfei/progressbar

and you can find example code for it here:

https://github.com/mikehearn/graviton-browser/blob/master/shell/src/main/kotlin/net/plan99/graviton/GravitonCLI.kt#L136